### PR TITLE
Joining and Leaving a Conversation

### PR DESCRIPTION
**Pull request by mariezimmy**
I had some hiccups with git, so if there are issues, let me know!

### CODE REVIEW

File: src/main/webapp/WEB-INF/view/chat.jsp [hunk 1]
```diff
@@ -80,20 +81,51 @@ List<Message> messages = (List<Message>) request.getAttribute("messages");
       </ul>
     </div>
 
+    <% if(request.getAttribute("error") != null){ %>
```
Comment (fatlotus) [inline src/main/webapp/WEB-INF/view/chat.jsp]: nit: spaces on the outside of ( )

File: src/main/java/codeu/controller/ChatServlet.java [hunk 2]
```diff
@@ -138,6 +142,23 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
...
+      conversation.conversationUsers.add(user);
+    }
+
+    if ("leaveButton".equals(button)) {
+      conversation.conversationUsers.remove(user);
+    }
+
+    if (!conversation.conversationUsers.contains(user)) {
+      // user has not joined conversation, don't let them add a message
+      response.sendRedirect("/chat/" + conversationTitle);
```
Comment (Mariamcm90) [inline src/main/java/codeu/controller/ChatServlet.java]: I'm not sure, but do we need this line?

File: src/main/webapp/WEB-INF/view/chat.jsp [hunk 3]
```diff
@@ -80,20 +81,51 @@ List<Message> messages = (List<Message>) request.getAttribute("messages");
...
+    </form>
+    <% } else { %>
+      <p><a href="/login">Login</a> to send a message.</p>
+    <% } %>
+
+    <hr/>
+
     <hr/>
 
     <% if (request.getSession().getAttribute("user") != null) { %>
```
Comment (Mariamcm90) [inline src/main/webapp/WEB-INF/view/chat.jsp]: If request.getSession().getAttribute("user") == null, I think the "Login to send a message" part will print twice. Maybe the former and latter sections can be combined.

File: src/main/webapp/WEB-INF/view/chat.jsp [hunk 4]
```diff
@@ -80,20 +81,51 @@ List<Message> messages = (List<Message>) request.getAttribute("messages");
...
     </form>
     <% } else { %>
       <p><a href="/login">Login</a> to send a message.</p>
     <% } %>
 
     <hr/>
 
+    <hr/>
+
+    <% if (request.getSession().getAttribute("user") != null) { %>
```
Comment (Mariamcm90) [inline src/main/webapp/WEB-INF/view/chat.jsp]: Also, I think maybe the leave button part can be combined with the first one so they can only leave a chat if they are a member of it already.

File: src/main/webapp/WEB-INF/view/chat.jsp (hunk 1 above)
Comment (mariezimmy) [reply src/main/webapp/WEB-INF/view/chat.jsp]: will fix then repush! Thanks!

File: src/main/webapp/WEB-INF/view/chat.jsp (hunk 3 above)
Comment (mariezimmy) [reply src/main/webapp/WEB-INF/view/chat.jsp]: Gotcha. Makes sense. Will fix then repush! Thanks, again!

**Comment by mariezimmy** [comment]
@Mariamcm90 let me know what you think about the chat.jsp, and if it looks good, I'll push to master!

**Comment by Mariamcm90** [comment]
Looks good to me! Thanks Marie :)

**Pull request merged**